aioshelly/rpc_device.py

```diff
@@ -172,32 +179,38 @@ def config(self):
         return self._config
 
     @property
-    def gen(self):
+    def gen(self) -> int:
         """Device generation: GEN2 - RPC."""
         return 2
 
     @property
-    def firmware_version(self):
+    def firmware_version(self) -> str:
         """Device firmware version."""
+        if self.shelly is None:
+            raise ShellyError
+
         if not self.initialized:
             raise NotInitialized
 
-        return self.shelly["fw_id"]
+        return str(self.shelly["fw_id"])
 
     @property
-    def model(self):
+    def model(self) -> str:
         """Device model."""
+        if self.shelly is None:
+            raise ShellyError
+
```

Comment:
Let's raise `AuthRequired` after we check if device is initialized same as we do for `device_info` and `config`
```suggestion
```